### ISSUE TITLE
`empty_line_after_outer_attr` doesn't seem to work for rustdoc comments

### ISSUE THREAD

**Issue by smoelius** [rust-lang/rust-clippy#10395]
### Summary

Pretty self-explanatory (I think).

### Lint Name

`empty_line_after_outer_attr`

### Reproducer

I tried this code ([playground link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=25dc51b4f387dc264a1b1d9fe8d35bb4)):

```rust
#![warn(clippy::empty_line_after_outer_attr)]

/// Main function

fn main() {}
```

I expected to see this happen:
A warning.

Instead, this happened:
No warning.

**Comment by jdswensen** [comment]
I took a [first pass at this](https://github.com/jdswensen/rust-clippy/commit/997f80462b71efd88933e894167cd85013f9a5b5), but wanted to make sure I'm on the right path before opening a PR. Two questions:

1. Is it desired that this lint encapsulate both outer attribute kinds? By that I mean, is it better as I've initially implemented it or would it be better to break the lint into two discrete configurations? The reason I ask is because of the next question.
2. [The tests currently fail on](https://github.com/rust-lang/rust-clippy/blob/8f0ba1f2e0c08d67c6b57a2e6b01cc87428b5341/tests/ui/empty_line_after_outer_attribute.rs#L56-L65):

```
/// This doc comment should not produce a warning

/** This is also a doc comment and should not produce a warning
 */

// This should not produce a warning
#[allow(non_camel_case_types)]
#[allow(missing_docs)]
#[allow(missing_docs)]
fn three_attributes() { assert!(true) }
```

By checking for `AttrKind::DocComment` in the outer attributes, `clippy` will end up catching cases like this. Is this new behavior acceptable? Or is there some other intended outcome? Making the change without introducing a different lint rule may fail code that currently has comments of this kind.

It doesn't look too difficult to break up. My plan was to have a helper function that we could pass `AttrKind` and `lint_msg` into as both lints could use the same existing logic. The new rule could the be something like `EMPTY_LINE_AFTER_OUTER_DOC_ATTR`.

Either way, let me know! Thanks!

**Comment by smoelius** [comment]
The lines you cited are from this commit: https://github.com/rust-lang/rust-clippy/commit/a64724fac49a91b1049f377f296b16a4725b713b

Apparently, the behavior I am complaining about was deliberate.

IMHO, lines like these _should_ produce a warning:
```
/// This doc comment should not produce a warning

```
If they appear near the top of the module, it begs whether the author meant to use an inner doc comment, for example:
```
//! This doc comment should not produce a warning

```
I would like to know that the Clippy team thinks, though.

Thanks very much for looking into this issue, @jdswensen.

**Comment by jdswensen** [comment]
That works for me! It's also the direction I was leaning after getting more familiar with the lint and test cases.
